### ISSUE TITLE
Adding bonus GBs to LDAP/AD synced user results in 500 error

### ISSUE THREAD

**Issue by brian-emery** [SpiderOak/netkes#208]
When adding bonus gigs to an LDAP/AD synced user and clicking 'save', the next page shows a 500 error. In the admin_console logs you see this:

```
INFO 2015-04-21 10:57:27,013 brian_t9 (superuser): edit user "jjames@example.com" with data: {'bonus_bytes': -9000000000}
INFO 2015-04-21 10:57:27,013 brian_t9 (superuser): edit user "jjames@example.com" with data: {'bonus_bytes': -9000000000}
2015-04-21 10:57:27,013 INFO    admin_actions  : brian_t9 (superuser): edit user "jjames@example.com" with data: {'bonus_bytes': -9000000000}
ERROR 2015-04-21 10:57:27,359 Internal Server Error: /users/jjames@example.com/
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 113, in get_response
    response = callback(request, *callback_args, **callback_kwargs)
  File "/opt/openmanage/django/apps/blue_management/blue_mgnt/views/views.py", line 473, in new_fun
    request.session['username'], *args, **kwargs)
  File "/opt/openmanage/django/apps/blue_management/blue_mgnt/views/users.py", line 456, in user_detail
    user_form.cleaned_data.get('email', email))
  File "/usr/local/lib/python2.7/dist-packages/django/shortcuts/__init__.py", line 78, in redirect
    return redirect_class(resolve_url(to, *args, **kwargs))
  File "/usr/local/lib/python2.7/dist-packages/django/shortcuts/__init__.py", line 151, in resolve_url
    return urlresolvers.reverse(to, args=args, kwargs=kwargs)
  File "/usr/local/lib/python2.7/dist-packages/django/core/urlresolvers.py", line 522, in reverse
    return iri_to_uri(resolver._reverse_with_prefix(view, prefix, *args, **kwargs))
  File "/usr/local/lib/python2.7/dist-packages/django/core/urlresolvers.py", line 442, in _reverse_with_prefix
    "arguments '%s' not found." % (lookup_view_s, args, kwargs))
NoReverseMatch: Reverse for 'user_detail_saved' with arguments '(u'',)' and keyword arguments '{}' not found.
ERROR 2015-04-21 10:57:27,359 Internal Server Error: /users/jjames@example.com/
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 113, in get_response
    response = callback(request, *callback_args, **callback_kwargs)
  File "/opt/openmanage/django/apps/blue_management/blue_mgnt/views/views.py", line 473, in new_fun
    request.session['username'], *args, **kwargs)
  File "/opt/openmanage/django/apps/blue_management/blue_mgnt/views/users.py", line 456, in user_detail
    user_form.cleaned_data.get('email', email))
  File "/usr/local/lib/python2.7/dist-packages/django/shortcuts/__init__.py", line 78, in redirect
    return redirect_class(resolve_url(to, *args, **kwargs))
  File "/usr/local/lib/python2.7/dist-packages/django/shortcuts/__init__.py", line 151, in resolve_url
    return urlresolvers.reverse(to, args=args, kwargs=kwargs)
  File "/usr/local/lib/python2.7/dist-packages/django/core/urlresolvers.py", line 522, in reverse
    return iri_to_uri(resolver._reverse_with_prefix(view, prefix, *args, **kwargs))
  File "/usr/local/lib/python2.7/dist-packages/django/core/urlresolvers.py", line 442, in _reverse_with_prefix
    "arguments '%s' not found." % (lookup_view_s, args, kwargs))
NoReverseMatch: Reverse for 'user_detail_saved' with arguments '(u'',)' and keyword arguments '{}' not found.
2015-04-21 10:57:27,359 ERROR   django.request : Internal Server Error: /users/jjames@example.com/
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 113, in get_response
    response = callback(request, *callback_args, **callback_kwargs)
  File "/opt/openmanage/django/apps/blue_management/blue_mgnt/views/views.py", line 473, in new_fun
    request.session['username'], *args, **kwargs)
  File "/opt/openmanage/django/apps/blue_management/blue_mgnt/views/users.py", line 456, in user_detail
    user_form.cleaned_data.get('email', email))
  File "/usr/local/lib/python2.7/dist-packages/django/shortcuts/__init__.py", line 78, in redirect
    return redirect_class(resolve_url(to, *args, **kwargs))
  File "/usr/local/lib/python2.7/dist-packages/django/shortcuts/__init__.py", line 151, in resolve_url
    return urlresolvers.reverse(to, args=args, kwargs=kwargs)
  File "/usr/local/lib/python2.7/dist-packages/django/core/urlresolvers.py", line 522, in reverse
    return iri_to_uri(resolver._reverse_with_prefix(view, prefix, *args, **kwargs))
  File "/usr/local/lib/python2.7/dist-packages/django/core/urlresolvers.py", line 442, in _reverse_with_prefix
    "arguments '%s' not found." % (lookup_view_s, args, kwargs))
NoReverseMatch: Reverse for 'user_detail_saved' with arguments '(u'',)' and keyword arguments '{}' not found.
```
When you navigate back to the user's page, the bonus seems to have been applied successfully. Local users seem unaffected by this issue:

```
INFO 2015-04-21 11:02:31,465 brian_t9 (superuser): edit user "spideroaktest+t9.user4@gmail.com" with data: {'name': u'user4 ', 'bonus_bytes': 3000000000, 'enabled': False, 'group_id': u'395', 'email': u'spideroaktest+t9.user4@gmail.com'}
INFO 2015-04-21 11:02:31,465 brian_t9 (superuser): edit user "spideroaktest+t9.user4@gmail.com" with data: {'name': u'user4 ', 'bonus_bytes': 3000000000, 'enabled': False, 'group_id': u'395', 'email': u'spideroaktest+t9.user4@gmail.com'}
2015-04-21 11:02:31,465 INFO    admin_actions  : brian_t9 (superuser): edit user "spideroaktest+t9.user4@gmail.com" with data: {'name': u'user4 ', 'bonus_bytes': 3000000000, 'enabled': False, 'group_id': u'395', 'email': u'spideroaktest+t9.user4@gmail.com'}
```

**Comment by bdzim** [comment]
Fixed in my development branch. This will go out in 2.2

**Comment by brian-emery** [comment]
Fixed in 2.2 release. Closing.
